### PR TITLE
Mechanistic modernisation of the Python code examples

### PR DESCRIPTION
**Pull request by PeterJCLaw**
Pretty much what it says. I've tried to keep this to the code examples.

The main things this does:
- drop using module-formatter (`%`) in favour of `str.format`
- used a context manager for file reading
- wrap some `range` calls in `list`; they're ugly now, but at least match the sample outputs
- change to print calls

See individual commit messages for my reasoning.

### CODE REVIEW

File: programming/python/functions.md
Comment (prophile) [inline programming/python/functions.md:36]: f-strings?

File: troubleshooting/index.md
Comment (prophile) [inline troubleshooting/index.md:45]: They aren't statements

**Review by prophile — APPROVED** [review]

File: programming/python/functions.md
Comment (PeterJCLaw) [reply programming/python/functions.md:36]: Hrm, didn't think about that. f-strings feel a _bit_ more magic here than `.format`. My goal was to fix things which were either now wrong or are foot-guns. In that light I'm going to leave this as-is, but happy for you to follow-on later and convert these further.

File: troubleshooting/index.md
Comment (PeterJCLaw) [reply troubleshooting/index.md:45]: Form the commit message:

> I've left in place the terminology of "print statements" because:
> a) it's teechnically still correct -- `print(x)` _is_ a statement
> b) this is easy to change later (at no additional effort) if we decide we really want to
> c) I think "print statement" sounds nicer than "print funtion call"

While `print` isn't a statement `print(x)` is.

Do you have a suggestion for a better terminology here?

**Review by sedders123 — APPROVED** [review]

**Comment by trickeydan** [comment]
This falls under the remit of the Kit Support Team. Please get their approval before merging.

**Comment by PeterJCLaw** [comment]
> This falls under the remit of the Kit Support Team. Please get their approval before merging.

As the simulator is Python 3, I see this as a critical PR for this year. I'm going to merge this before Kickstart starts, but if they'd like to input before then they're welcome to.

**Pull request merged**
